es6-lib/errors/index.js

```diff
@@ -0,0 +1,110 @@
+import _ from 'underscore';
+import changeCase from 'change-case';
+
+class GeoError extends Error {
+
+  constructor(parameters) {
+    super();
+    this._params = parameters || {};
+    if (!_.every(this.required, (key) => !!this._params[key])) {
+      throw new Error(`Missing required parameter for ${this.t} error!`);
+    }
+  }
+
+  toString() {
+    return JSON.stringify(this.toJSON());
+  }
+
+  get t() {
+    return changeCase.snakeCase(this.constructor.name);
+  }
+
+  toJSON() {
+    return _.extend({
+      'type': this.t,
+    }, this.parameters);
+  }
+
+  get parameters() {
+    return this._params;
+  }
+
+  get statusCode() {
+    return 400;
+  }
+}
+
+
+class UninterpretableValue extends GeoError {
+  get required() {
+    return ['record', 'filename'];
+  }
+}
+
+class CorruptShapefile extends GeoError {
+  get required() {
+    return ['filename'];
+  }
+}
+
+class CorruptArchive extends GeoError {
+  get required() {
+    return ['reason'];
+  }
+}
+
+class IncompleteArchive extends GeoError {
+  get required() {
+    return ['missing'];
+  }
+}
+
+class ParseError extends GeoError {
+  get required() {
+    return ['lineNumber', 'reason'];
+  }
+}
+
+class RecordTooLarge extends GeoError {
+  get required() {
+    return ['limit', 'nearRecord'];
+  }
+}
+
+class UnknownFileType extends GeoError {
+  get required() {
+    return ['extension'];
```

Comment:
I might be being a little pedantic, but given how it's used `extension` seems a little misleading.  Maybe it would be better as `contentType`?  Or perhaps changing the error to `UnknownMimeType` and changing the field to `type` would be better?